cmd/fluxctl/args.go

```diff
@@ -16,11 +19,64 @@ func parseServiceOption(s string) (update.ServiceSpec, error) {
 }
 
 func AddCauseFlags(cmd *cobra.Command, opts *update.Cause) {
-	username := ""
-	user, err := user.Current()
-	if err == nil {
-		username = user.Username
-	}
+	authorInfo := getUserGitconfig()
+	username := getCommitAuthor(authorInfo)
+
 	cmd.Flags().StringVarP(&opts.Message, "message", "m", "", "attach a message to the update")
-	cmd.Flags().StringVar(&opts.User, "user", username, "override the user reported as initating the update")
+	cmd.Flags().StringVar(&opts.User, "user", username, "override the user reported as initiating the update")
+}
+
+func getCommitAuthor(authorInfo map[string]string) string {
+	userName := authorInfo["user.name"]
+	userEmail := authorInfo["user.email"]
+
+	switch userName {
```

Comment:
This is the equivalent of a nested if. I think the table of results is better expressed as

```
switch {
case userEmail != "" && userName != "":
  return fmt.Sprintf("%s <%s>", userName, userEmail)
case userEmail != "":
  return userEmail
case userName != "":
  return userName
}
return ""
```